File: ver_cpp/T09.cpp
// https://leetcode.cn/problems/power-of-four/
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    bool isPowerOfFour(int n)
    {
        if (n <= 0)
            return false;
        if (n == 1)
            return true;
        long long ans = 1;
        while (ans < n)
        {
            ans *= 4;
            if (ans == n)
                return true;
        }
        return false;
    }
};

int main()
{
    Solution s;
    long long n;
    cin >> n;
    cout << s.isPowerOfFour(n) << endl;
    return 0;
}

File: ver_py/T09.py
#  https://leetcode.cn/problems/power-of-four/
class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        ans = 1
        if n <= 0:
            return False
        elif n == 1:
            return True
        else:
            while ans < n:
                ans *= 4
            if ans == n:
                return True
            else:
                return False
if __name__ == "__main__":
    n = int(input())
    print(Solution().isPowerOfFour(n))